llvm/lib/AsmParser/LLParser.cpp

```diff
@@ -6216,6 +6225,11 @@ bool LLParser::convertValIDToValue(Type *Ty, ValID &ID, Value *&V,
     V = ConstantVector::getSplat(cast<VectorType>(Ty)->getElementCount(),
                                  ID.ConstantVal);
     return false;
+  case ValID::t_MDString:
```

Comment:
Instead of adding this here, shouldn't we be calling parseMetadataAsValue() for metadata types in parseOptionalOperandBundles? That way we'd match how call arguments are parsed.

Reply:
This is determined by the logic of `parseOptionalOperandBundles`. It parses type then value:
https://github.com/llvm/llvm-project/blob/68c04b0ae62d8431d72d8b47fc13008002ee4387/llvm/lib/AsmParser/LLParser.cpp#L3205

Parsing value involves first parsing to `ValID`, then conversion to the value: https://github.com/llvm/llvm-project/blob/68c04b0ae62d8431d72d8b47fc13008002ee4387/llvm/lib/AsmParser/LLParser.cpp#L6274-L6279

Processing metadata values was implemented according to this logic.

Reply:
Changed `parseOptionalOperandBundles` accordingly.